include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -124,16 +150,20 @@ struct __subgroup_radix_sort
         return __req_slm_size_val <= __max_slm_size - __req_slm_size_counters; //counters should be placed in SLM
     }
 
-    template <typename _KernelName, typename _SLM_tag, typename _RangeIn>
+    template <typename _KernelName, typename _SLM_tag, typename _RangeIn, typename _Proj>
     auto
-    __submit(sycl::queue __q, _RangeIn&& __src)
+    __submit(sycl::queue __q, _RangeIn&& __src, _Proj __proj)
     {
         uint16_t __n = __src.size();
         assert(__n <= __block_size * __wg_size);
 
-        using _KeyT = oneapi::dpl::__internal::__value_t<_RangeIn>;
+        using _ValT = oneapi::dpl::__internal::__value_t<_RangeIn>;
+        using _KeyT = oneapi::dpl::__internal::__key_t<_Proj, _RangeIn>;
+
+        //A value is a key: we use an uniform iteration space and use a defaultult(identical) value for __i >= __n
+        constexpr bool __uniform_space = std::is_same_v<_KeyT, _ValT> && sizeof(_KeyT) == sizeof(_ValT);
```

Comment:
`&& sizeof(_KeyT) == sizeof(_ValT)`  is this necessary?  If they are the same, shouldn't it always be true?

Reply:
yea..